Code/Framework/AzCore/AzCore/Debug/Profiler.inl

```diff
@@ -14,44 +14,44 @@ namespace AZ::Debug
     void ProfileScope::BeginRegion(
         [[maybe_unused]] Budget* budget, [[maybe_unused]] const char* eventName, [[maybe_unused]] T const&... args)
     {
-        if (!budget)
+    #if !defined(_RELEASE)
+        if (budget)
         {
-            return;
-        }
-#if !defined(_RELEASE)
-        // TODO: Verification that the supplied system name corresponds to a known budget
-#if defined(USE_PIX)
-        PIXBeginEvent(PIX_COLOR_INDEX(budget->Crc() & 0xff), eventName, args...);
-#endif
-        budget->BeginProfileRegion();
+        #if defined(USE_PIX)
+            PIXBeginEvent(PIX_COLOR_INDEX(budget->Crc() & 0xff), eventName, args...);
+        #endif // defined(USE_PIX)
```

Comment:
I'm concerned about all these special cases for PIX being added. There's a common interface that can be used to send this information over so PIX should not be a special case but instead hook into that interface just like any other profiler would, especially since PIX only applies to one or two platforms.

Reply:
These aren't being added but moved around.  There is a bit of overlap with this kind of clean up in #7021, but you'll at least see in that change these platform specific API calls get properly abstracted.